src/Consolonia.Core/Infrastructure/DefaultNetConsoleOutput.cs

```diff
@@ -111,6 +115,19 @@ public virtual void PrepareConsole()
             _originalForeground = Console.ForegroundColor;
             _originalBackground = Console.BackgroundColor;
             Size = new PixelBufferSize((ushort)Console.WindowWidth, (ushort)Console.WindowHeight);
+            // Detect complex emoji support by writing a complex emoji and checking cursor position.
+            // If the cursor moves 2 positions, it indicates proper rendering of composite surrogate pairs.
+            (int left, _) = Console.GetCursorPosition();
+            WriteText("👨‍👩‍👧‍👦");
```

Comment:
i guess for this printing we should set foreground to be same as background. This way user will not see this internal this easy

Reply:
I tried that, but foreground is ignored for complex emojii rendering. If we can find one which is multicodepoint that renders using text color than  I agree.